common/src/main/java/com/wynntils/core/notifications/NotificationManager.java

```diff
@@ -29,6 +43,26 @@ public static MessageContainer queueMessage(TextRenderTask message) {
 
         WynntilsMod.info("Message Queued: " + message);
         MessageContainer msgContainer = new MessageContainer(message);
+        String messageText = message.getText();
+
+        for (Pair<String, MessageContainer> cachedMessagePair : cachedMessageSet) {
+            String checkableMessage = cachedMessagePair.a();
+            if (messageText.equals(checkableMessage)) {
+                MessageContainer matchedContainer = cachedMessagePair.b();
+                matchedContainer.incrementMessageCount();
+
+                // Overlay is not enabled, send in chat
+                if (!GameNotificationOverlayFeature.INSTANCE.isEnabled()) {
+                    sendOrEditNotification(msgContainer);
+                }
+
+                WynntilsMod.postEvent(new NotificationEvent.Edit(matchedContainer));
+                return matchedContainer;
+            }
+        }
+
+        cachedMessageSet.put(new Pair<>(messageText, msgContainer));
+        msgContainer.incrementMessageCount();
```

Comment:
I see! So you default to 0, increment instantly. That is a bit of a twisted logic though, imo. Let's just start it from 1 and remove this line.